src/main/java/kitchenpos/order/application/OrderTableValidatorImpl.java

```diff
@@ -0,0 +1,36 @@
+package kitchenpos.order.application;
+
+import java.util.List;
+import java.util.Optional;
+import kitchenpos.order.Order;
+import kitchenpos.order.repository.OrderRepository;
+import kitchenpos.ordertable.application.OrderTableValidator;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OrderTableValidatorImpl implements OrderTableValidator {
+
+    private final OrderRepository orderRepository;
+
+    public OrderTableValidatorImpl(final OrderRepository orderRepository) {
+        this.orderRepository = orderRepository;
+    }
+
+    @Override
+    public void checkOrderComplete(final Long orderTableId) {
+        final Optional<Order> order = orderRepository.findByOrderTableId(orderTableId);
+        if (order.isPresent() && !order.get().isComplete()) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    @Override
+    public void checkOrderComplete(final List<Long> orderTableIds) {
+        final List<Order> orders = orderRepository.findByOrderTableIdIn(orderTableIds);
+        final boolean isAllComplete = orders.stream()
+                .allMatch(Order::isComplete);
+        if (!isAllComplete) {
+            throw new IllegalArgumentException();
+        }
```

Comment:
```suggestion
        final boolean isNotAllComplete = orders.stream()
                .noneMatch(Order::isComplete);
        if (isNotAllComplete) {
            throw new IllegalArgumentException();
        }
```

`noneMatch()` 를 사용해볼 수 있을 것 같아요!

Reply:
후디 스트림 고수시군요 👍 감사합니다!!